### PR TITLE
Add compatibility with remote_config 5.0.0

### LINKED ISSUE #3: Update remote config 5.x,  firebase core 4.x
```
Because fake_firebase_remote_config <1.0.2 depends on firebase_remote_config >=0.9.0-dev.0 <2.0.0 and fake_firebase_remote_config >=1.0.2 <1.0.3 depends on firebase_remote_config >=0.9.0-dev.0 <3.0.0, fake_firebase_remote_config <1.0.3 requires firebase_remote_config >=0.9.0-dev.0 <3.0.0.
And because fake_firebase_remote_config >=1.0.3 depends on firebase_core >=0.0.1 <3.0.0 and insight_app depends on firebase_core ^3.0.0, every version of fake_firebase_remote_config requires firebase_remote_config >=0.9.0-dev.0 <3.0.0.
So, because insight_app depends on both firebase_remote_config ^5.0.0 and fake_firebase_remote_config any, version solving failed.
exit code 1
```

### PR DESCRIPTION
**Pull request by dfdgsdfg**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ environment:
   flutter: ">=1.17.0"
 
 dependencies:
-  firebase_core: ">= 0.0.1 <3.0.0"
-  firebase_remote_config: ">=4.0.0 <5.0.0"
+  firebase_core: ">= 0.0.1 <4.0.0"
+  firebase_remote_config: ">=5.0.0 <6.0.0"
```
Comment (ciriousjoker) [inline pubspec.yaml]: Given that only the dependency requirements change, but no actual code, shouldn't the lower boundary (4.0.0) be unchanged?

Comment (dfdgsdfg) [reply pubspec.yaml]: https://github.com/firebase/flutterfire/blob/master/packages/firebase_remote_config/firebase_remote_config/CHANGELOG.md#500

---

5.0.0
Note: This release has breaking changes.

BREAKING REFACTOR: android plugins require minSdk 21, auth requires minSdk 23 ahead of android BOM >=33.0.0 (https://github.com/firebase/flutterfire/issues/12873). ([52accfc6](https://github.com/firebase/flutterfire/commit/52accfc6c39d6360d9c0f36efe369ede990b7362))
BREAKING REFACTOR: bump all iOS deployment targets to iOS 13 ahead of Firebase iOS SDK v11 breaking change (https://github.com/firebase/flutterfire/issues/12872). ([de0cea2c](https://github.com/firebase/flutterfire/commit/de0cea2c3c36694a76361be784255986fac84a43))

---

I agree. Only dependencies changed.

Comment (ciriousjoker) [reply pubspec.yaml]: In that case I'd leave the lower boundary as-is. The minSdk requirement is automatically enforced during build.

**Review by ciriousjoker — APPROVED** [review]
lgtm, I just bumped down the version, since 3.0.0 would mean this is breaking, but it's not.

**Comment by ciriousjoker** [comment]
@dfdgsdfg Sorry for taking so long and thanks for your efforts!

**Pull request merged**